recipes/cpr/all/conanfile.py

```diff
@@ -36,7 +36,7 @@ def configure(self):
             del self.options.fPIC
         if self.options.with_openssl:
             # If using OpenSSL, we need it to be active in libcurl too
-            self.options["libcurl"].with_openssl = True
+            self.options["libcurl"].with_ssl = "openssl"
```

Comment:
I think this should be moved to `build` and changed in a check:
```python
if not self.options["libcurl"].with_openssl != "openssl":
    raise ConanInvalidConfiguration("cpr requires libcurl built with with_ssl=\"openssl\"")
```

Reply:
Makes more sense!

Reply:
I disagree, because `build()` is not always called, and you probably want that recipe works out of the box.

Reply:
Then the check should be called in `package_info`.

Reply:
Yes maybe, but anyway I believe that upstream option should be forced if it's a strong requirement of this recipe. What happen if default option of libcurl is not `with_ssl = openssl"`? No binary is produced in CCI?

Reply:
Forcing of options is not correct as it will override a user passing e.g. `-o libcurl:with_ssl="gnutls"`.
This recipe is not relying on libcurl default options as it actively checks the option.

Reply:
It's even more surprising for a user to have a recipe not building with its default options. This recipe relies on libcurl default option to build out of the box.

If you just launch `conan install somerecipe/x.x.x` and it raises because of some options in transitive dependencies, it's quite surprising.

Depending on recipe design, an option could be a feature, or a component which could has been its own recipe.

In conan 2.0, precedence of values should change https://github.com/conan-io/conan/issues/7786

Reply:
On the same topic:
- in old and new `libcurl` recipe (without and with `with_ssl` option): the default value for ssl lib on Macos is `darwinssl`
- in old libcurl recipe, forcing `with_openssl=True` didn't have any effect when `darwin_ssl` was already True. So CPR was somehow broken, if openssl is strictly required.
- in new libcurl recipe, forcing `with_ssl =openssl` will indeed force using `openssl` in `libcurl`. But it's not the default value on Macos, so there is no pre build binaries in CCI for such package ids.
- cpr recipe claims that openssl support is unstable, so why not disabling it by default?

Reply:
Okay, let's keep forcing option.